roles/rsyslog/tasks/main.yaml

```diff
@@ -1,43 +1,34 @@
----
 - name: Install/Update required packages
   package:
-    name: "{{ lookup('flattened', rsyslog_base_packages, rsyslog_tls_packages if rsyslog_pki else [], rsyslog_packages) }}"
+    name: "{{ (__rsyslog_base_packages) + (__rsyslog_tls_packages if rsyslog_pki else []) + (rsyslog_extra_packages | flatten) }}"
     state: latest
   when:
     - rsyslog_enabled | bool
-    - not use_rsyslog_image | default(false) | bool
-  ignore_errors: yes
-  tags:
-    - skip_ansible_lint
+
+- name: Gather package facts
+  package_facts:
+    manager: auto
 
 - name: Get rsyslog version
-  package:
-    list: rsyslog
-  register: rsyslog_version
+  set_fact:
+    __rsyslog_version: >-
+      {{ ansible_facts.packages['rsyslog'][0].version
+      if ansible_facts.packages | selectattr('rsyslog', 'defined') | list | length > 0
+      else '0.0.0' }}
```

Comment:
do you need set_fact or register? Can't you use vars: only on the block or tasks where the variables are actually used?

Reply:
I would say it is using `ansible_facts.packages`. But my comment was not about the way to obtain the version, but about what is being done with the value. I don't think a fact or registered variable should be used.

Reply:
@pcahyna, currently, __rsyslog_version is referred just in roles/rsyslog/tasks/main.yaml.  But I'm anticipating the value might be used in some other task files to absorb the version differences in the future.  What type of variable would be most appropriate for such a purpose?

Reply:
> What type of variable would be most appropriate for such a purpose?

I'm not sure, but let's do that in a separate commit in the future, when such functionality is required.

Reply:
Ok.  So, is it ok to keep this change in this PR or should I revert it?
https://github.com/linux-system-roles/logging/pull/87/files#diff-ed5a08a2e61430dc18e2d3820cbd6f32L14-R17

Reply:
Hmm - for some reason that link isn't working for me - what is that change?

Reply:
Sorry, this is the raw diff.  (indeed, the link does not work for me, either ):
Actually, I realized there were 2 issues.  In the beginning, rsyslog version was retrieved using the yum module [1].  But we are to use `package` which is a higher level, platform dependent module.  Unfortunately, it does not return version.  So, I fixed it with the ansible_facts data [2].  That being said, we do not want to go back to `yum:`?

[1]
```
 - name: Get rsyslog version
-  yum:
+  package:
     list: rsyslog
   register: rsyslog_version
```
[2]
```
 - name: Get rsyslog version
-  package:
-    list: rsyslog
-  register: rsyslog_version
+  set_fact:
+    __rsyslog_version: >-
+      {{ ansible_facts.packages['rsyslog'][0].version
+      if ansible_facts.packages | selectattr('rsyslog', 'defined') | list | length > 0
+      else '0.0.0' }}
```

Reply:
no, we do not want to go back to `yum` 
as pcahyna said, using `ansible_facts.packages['rsyslog'][0].version` is the canonical way to get the package version.

Reply:
By the way, do you really need `| length > 0` ?

Reply:
> By the way, do you really need `| length > 0` ?

@pcahyna, no, it's not needed.

Reply:
@pchayna
> By the way, do you really need `| length > 0` ?

I've opened a pr to remove `| length > 0`.  Could you please take a look?
https://github.com/linux-system-roles/logging/pull/91
Also, it turned out selectattr did not work for this case [1], thus I replaced it with json_query.  But if you think I should use the timesync method https://github.com/linux-system-roles/timesync/pull/60, please add your comment to https://github.com/linux-system-roles/logging/pull/91.  I'm going to update the pr.  Thanks.

[1] - Debug outputs from `ansible_facts.packages | json_query('rsyslog')` and `ansible_facts.packages | selectattr('rsyslog', 'defined') | list`.
```
TASK [linux-system-roles/logging/roles/rsyslog : debug] *******************************************
task path: linux-system-roles/logging/roles/rsyslog/tasks/main.yaml:12
ok: [momo7.localdomain] => {
    "msg": "json_query - [{'name': 'rsyslog', 'source': 'rpm', 'epoch': None, 'version': '8.24.0', 'release': '41.el7_7.2', 'arch': 'x86_64'}]"
}

TASK [linux-system-roles/logging/roles/rsyslog : debug] *******************************************
task path: linux-system-roles/logging/roles/rsyslog/tasks/main.yaml:15
ok: [momo7.localdomain] => {
    "msg": "selectattr - []"
}
```

Reply:
maybe I already said this but timesync now uses an even simpler approach: https://github.com/linux-system-roles/timesync/pull/67/files#diff-6d62b73b3f46cfa11f6466d08cf45303